src/databricks/labs/lakebridge/config.py

```diff
@@ -259,19 +259,32 @@ class ReconcileMetadataConfig:
     volume: str = "reconcile_volume"
 
 
+@dataclass
+class ReconcileCredentialConfig:
+    vault_type: str  # supports local, env, databricks creds.
+    source_creds: dict[str, str]
```

Comment:
I think we need a better name for this: it doesn't hold the credentials… it's more of a vault configuration?